### ISSUE TITLE
Use release tags for continued development each year

### ISSUE THREAD

**Issue by Glavin001** [SMUhackaweek/SMU-Hack-A-Week-2015#2]
I'm not quite sure why we do not simply create a release tag for last year and then continue development on previous repository. Even if the changes are dramatic, it seems strange to have SMU-Hack-A-Week-#{YEAR} for every year when we are using *Git*.

Just my spare twelve cents. Guess it's too late now anyway....

**Comment by valencik** [comment]
Perhaps I don't see the motivation of using release tags. What benefit does it bring?
Each year is a different event, this year was a fork of last year's site but that may not always be the case.
This approach also has the benefit of being able to simultaneously host each year on GitHub.

**Comment by Glavin001** [comment]
When MongoDB decided to develop 3.0 I highly doubt they forked their existing MongoDB project (would that repo be named mongodb-2.0 then?) and continued on a separate repository.. 
Why use release tags? Simply put: because that's how it's done and this use case is exactly why they exist.

> Perhaps I don't see the motivation of using release tags. What benefit does it bring?

If we are working on a project *based off of last year's repo*, then why not carry over all of those issues? They still exist, they are still applicable -- UI bugs, added features, etc -- just that the content will change -- which is expected as part of this *new release*.

> This approach also has the benefit of being able to simultaneously host each year on GitHub.

As you know, every commit is hosted, each year, on GitHub and release tags are like bookmarks for viewing them, accomplishing this "simultaneously [hosting of] each year [of the project] on GitHub".  That's what release tags are for: having a single project that is evolving over time and being able to keep track of versions, or other milestones (year), and go back and view them as it was at any time.

> Each year is a different event, this year was a fork of last year's site but that may not always be the case.

*My assumption as that each year would effectively be a fork of last year's site, however if that is not the case then that makes this void.* However, I would suggest that we *do* continue developing upon our previous/existing version of the site and continue improving our SMU-style hackathon site, instead of starting from scratch / simply ripping off from someone else's (McHacks).

**Comment by valencik** [comment]
Excellent points and thanks for the discussion.

I left out a very important word here:
> This approach also has the benefit of being able to simultaneously host each year on GitHub __Pages__.

Your point about the issues still being applicable is particularly convincing.
@kashifkai28 any thoughts on the matter?

I do not think we are beyond the point of combining the two repos. Although migrating issues is nontrivial.

**Comment by Glavin001** [comment]
Simultaneous GitHub Pages is a neat idea.

> I do not think we are beyond the point of combining the two repos. Although migrating issues is nontrivial.

I'd just let the old repo be the main (no new issues here) and then change the remote url from this to the main repo and push. Fortunately, the commit tree is the same upstream, so we can do this.

**Comment by kashifkai28** [comment]
Yes, I guess using a release tag would be good as long as are building on the older versions. I dont think the future hackathon themes would be entirely different from the previous ones, so we should be able to use releases. (Simultaneous GitHub pages, a bonus!) 
However, if the hackathon events are gonna be dramatically different, then its a different case.
Final Thoughts?

**Comment by Glavin001** [comment]
There are some projects I know of that have created new repos for their 2.0, however that is because they literally rewrote it from scratch. If we intend on continue from previous years' development (sounds like a yes) then let's just have one nice repo.

I'd say point your remotes at the old repo and push / submit a pull request there, and let's continue from there.
